source/common/conn_pool/conn_pool_base.cc

```diff
@@ -533,6 +539,14 @@ void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view
       onUpstreamReady();
     }
     checkForIdleAndCloseIdleConnsIfDraining();
+    break;
+  }
+  case Network::ConnectionEvent::ConnectedZeroRtt: {
+    ASSERT(client.state() == ActiveClient::State::CONNECTING);
+    host()->cluster().stats().upstream_cx_connect_with_0_rtt_.inc();
+    client.allows_early_data_ = true;
```

Comment:
I suspect this is correct, but I'm a bit surprised by this. I would have thought that "allows_early_data_" would be, for example, the property of a request (Safe vs not). But in this case, we're setting it to true when the connection uses 0-RTT. Is the idea that this means "if this client sent data now it would be over 0-RTT"?

Reply:
If we want to mean "this client can be used to send early data" then I would probably think "can_use_early_data" (or some such).